pkl-parser/src/main/java/org/pkl/parser/Parser.java

```diff
@@ -1393,6 +1393,10 @@ private Type parseTypeAtom(@Nullable String expectation) {
           children.add(ret);
           typ = new Type.FunctionType(children, tk.span.endWith(ret.span()));
         } else {
+          if (children.isEmpty()) {
+            backtrack();
+            throw parserError("unexpectedTokenForType", ")");
+          }
```

Comment:
```suggestion
          if (children.isEmpty()) {
            throw parserError("unexpectedTokenForType", ")");
          }
```
You don't need to backtrack if you are going to throw an error.

Reply:
Makes sense! Made that adjustment.